packages/mui-base/src/useSwitch/useSwitch.ts

```diff
@@ -51,79 +51,46 @@ export function useSwitch(props: UseSwitchParameters): UseSwitchReturnValue {
       otherProps.onChange?.(event);
     };
 
-  const {
-    isFocusVisibleRef,
-    onBlur: handleBlurVisible,
-    onFocus: handleFocusVisible,
-    ref: focusVisibleRef,
-  } = useIsFocusVisible();
-
-  const [focusVisible, setFocusVisible] = React.useState(false);
-  if (disabled && focusVisible) {
-    setFocusVisible(false);
-  }
-
-  React.useEffect(() => {
-    isFocusVisibleRef.current = focusVisible;
-  }, [focusVisible, isFocusVisibleRef]);
-
-  const inputRef = React.useRef<HTMLInputElement | null>(null);
-
-  const createHandleFocus =
-    (otherProps: React.InputHTMLAttributes<HTMLInputElement>) =>
-    (event: React.FocusEvent<HTMLInputElement>) => {
-      // Fix for https://github.com/facebook/react/issues/7769
-      if (!inputRef.current) {
-        inputRef.current = event.currentTarget;
-      }
-
-      handleFocusVisible(event);
-      if (isFocusVisibleRef.current === true) {
-        setFocusVisible(true);
-        onFocusVisible?.(event);
-      }
-
-      onFocus?.(event);
-      otherProps.onFocus?.(event);
-    };
-
-  const createHandleBlur =
-    (otherProps: React.InputHTMLAttributes<HTMLInputElement>) =>
-    (event: React.FocusEvent<HTMLInputElement>) => {
-      handleBlurVisible(event);
-
-      if (isFocusVisibleRef.current === false) {
-        setFocusVisible(false);
+  const createHandleClick =
+    (otherProps: React.ButtonHTMLAttributes<HTMLButtonElement>) =>
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      otherProps.onClick?.(event);
+      if (event.defaultPrevented || readOnly) {
+        return;
       }
 
-      onBlur?.(event);
-      otherProps.onBlur?.(event);
+      inputRef.current?.click();
     };
 
-  const handleInputRef = useForkRef(focusVisibleRef, inputRef);
+  const getButtonProps: UseSwitchReturnValue['getButtonProps'] = (otherProps = {}) => ({
+    ...otherProps,
+    type: 'button',
+    role: 'switch',
+    'aria-checked': checked,
+    'aria-disabled': disabled,
+    'aria-readonly': readOnly,
+    onClick: createHandleClick(otherProps),
+  });
 
   const getInputProps: UseSwitchReturnValue['getInputProps'] = (otherProps = {}) => ({
-    checked: checkedProp,
-    defaultChecked,
+    checked,
     disabled,
-    readOnly,
-    ref: handleInputRef,
+    name,
     required,
+    style: visuallyHidden,
+    tabIndex: -1,
     type: 'checkbox',
-    role: 'switch',
-    'aria-checked': checkedProp,
+    'aria-hidden': true,
     ...otherProps,
+    ref: handleInputRef,
     onChange: createHandleInputChange(otherProps),
-    onFocus: createHandleFocus(otherProps),
-    onBlur: createHandleBlur(otherProps),
   });
 
   return {
     checked,
     disabled: Boolean(disabled),
-    focusVisible,
+    getButtonProps,
     getInputProps,
-    inputRef: handleInputRef,
     readOnly: Boolean(readOnly),
   };
```

Comment:
I think we should only return `checked`, `getButtonProps`, and `getInputProps` since the other props are passed in as params to the hook, so they'll already have those values externally.

Reply:
Also, should set defualt values in the destructure `props` call, rather than use `Boolean()`?